src/Decompiler/Decompiler.cs

```diff
@@ -668,14 +547,15 @@ private void WriteEntryPoints(Program program)
         {
             if (program.Platform == null)
                 return;
-            var irPath = Path.Combine(program.SourceDirectory, Path.GetFileNameWithoutExtension(program.Filename));
+            var irPath = Path.Combine(program.SourceDirectory, Path.GetFileNameWithoutExtension(program.Location.FilesystemPath));
             program.Platform.WriteMetadata(program, irPath);
         }
 
 		public void WriteHeaderComment(string filename, Program program, TextWriter w)
 		{
+            var fragment = program.Location.FilesystemPath;
 			w.WriteLine("// {0}", filename);
-			w.WriteLine("// Generated by decompiling {0}", Path.GetFileName(program.Filename));
+			w.WriteLine("// Generated by decompiling {0}", Path.GetFileName(fragment));
```

Comment:
Looks like it should be `program.Location.GetFilename()`

Reply:
Fixed.